README.md

```diff
@@ -41,56 +42,58 @@ OpenChatML is intended to be used in various conversational AI scenarios, includ
 
 ## 3. Message Structure
+
 Each message in OpenChatML is represented as follows:
 
 ```
-<|im_start|>role [name=<name>]
-message_content
-<|im_end|>
+<|im_start|>role [<name>]
+message_content<|im_end|>
 ```
 
-- `role`: A string indicating the role of the speaker. It must be one of the following: "system", "tool", "user", or "assistant".
-- `name` (optional): A string representing the name of the speaker. If present, it should be added after the role, in the format `name=<name>`. The name cannot contain whitespace.
+- `role`: A string indicating the role of the speaker. It must be one of the following: "system", "tool", "user", or "char".
+- `name` (optional): A string representing the name of the speaker. If present, it should be added after the role, in the format `<name>`. The name cannot contain whitespace.
 - `message_content`: The actual content of the message, which can span multiple lines.
 
 ## 4. Thought Structure
-OpenChatML introduces a new structure called the "thought block" to represent the chain of thought or reasoning steps that lead to a conclusion or response. This concept is inspired by the Quiet-STaR paper (Zelikman et al., 2022), which proposes a method for language models to generate rationales at each token to explain future text, improving their predictions. In OpenChatML, the thought block is enclosed within <|startofthought|> and <|endofthought|> tokens and contains the intermediate reasoning steps or considerations that the model uses to arrive at its final response. The purpose of the thought block is to provide insight into the model's decision-making process and to separate the reasoning from the conclusion. This structure allows for a clearer understanding of how the model generates its responses and can be useful for debugging, interpretability, and enhancing the model's performance on various tasks.
+
+OpenChatML introduces a new structure called the "thought block" to represent the chain of thought or reasoning steps that lead to a conclusion or response. This concept is inspired by the Quiet-STaR paper (Zelikman et al., 2022), which proposes a method for language models to generate rationales at each token to explain future text, improving their predictions. In OpenChatML, the thought block is enclosed within `<|start_of|>thought` and `<|end_of|>thought` tokens and contains the intermediate reasoning steps or considerations that the model uses to arrive at its final response. The purpose of the thought block is to provide insight into the model's decision-making process and to separate the reasoning from the conclusion. This structure allows for a clearer understanding of how the model generates its responses and can be useful for debugging, interpretability, and enhancing the model's performance on various tasks.
 
 Thoughts or rationales in OpenChatML are represented as follows:
 
 ```
-<|startofthought|>thought_content<|endofthought|>
+<|start_of|>thought thought_content<|end_of|>thought
```

Comment:
I like the concept, but really it's better to train the model on token tags rather than strings.